### ISSUE TITLE
[feature]: documentation of environment vars

### ISSUE THREAD

**Issue by stefanux** [makeplane/plane#4649]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

content of .env-Files is not documented (at least i did not find any).

with v20 new vars were introduced:
WARN[0000] The "ADMIN_BASE_URL" variable is not set. Defaulting to a blank string.
WARN[0000] The "SPACE_BASE_URL" variable is not set. Defaulting to a blank string.
WARN[0000] The "APP_BASE_URL" variable is not set. Defaulting to a blank string.

please create/update documentation with alle existing vars plus their default values (if not set).

### Why should this be worked on?

because its needed for selfhosted users and/or making a ansible role

**Comment by sriramveeraghanta** [comment]
**Future use case:** These variables are not currently in use because the application is served under a reverse proxy, allowing it to function as a single application despite multiple running services. These environment variables will be necessary when hosting the self-hosted version across various subdomains, with each service assigned to its subdomain. We will add the required documentation when we are releasing this provision.

**Comment by stefanux** [comment]
> These variables are not currently in use because the application is served under a reverse proxy
do you mean the named three new ones (ADMIN_BASE_URL, SPACE_BASE_URL, APP_BASE_URL)?

the main point of this ticket is the documentation of the existing vars. selfhosting should always be a first class citizen (even if it SaaS might be more profitable).

**Comment by theparthacus** [comment]
yeah, i hear you. the existing variables are documented. if you find a miss or something doesn't look right, please let us know and we will fix it. the three variables that you are talking about in this thread are, as mentioned above, for the future and we will document them when we make provisions for their use.

does that make sense?

**Comment by stefanux** [comment]
if the vars are future use, than i personally would tone them down to not make prominent warnings at startup (or at least change the text of the warning).

if you have a deeplink to the proper documentation, i would be happy if you share it. i found the examples ( https://github.com/makeplane/plane/blob/preview/ENV_SETUP.md ) but no explanations.

**Comment by sarim** [comment]
@theparthacus 
I also got confused about these three new variables without any reference and getting warning from docker about these. I ended up adding them as empty in my env file.

I even run setup.sh in a fresh directory again to make it generate a fresh env file. There's no mention of these variables there either, I think that's a bug. If you add this to env many confusions will be solved.

```properties
# Add an appropriate description, ex:
# For future use. Can be kept empty in singular instance
ADMIN_BASE_URL=
SPACE_BASE_URL=
APP_BASE_URL=
```

**Comment by theparthacus** [comment]
@sriramveeraghanta, this is a nice suggestion. let's comment it out for now, i guess.

**Comment by vincent067** [comment]
Hi everyone! 👋

I just came across this issue while setting up Plane locally. The confusion around ADMIN_BASE_URL, SPACE_BASE_URL, and APP_BASE_URL is real - those Docker warnings can be misleading for new self-hosted users.

I really like @sarim's suggestion of adding them to the env file with comments explaining they're for future use. This would definitely improve the self-hosting experience! 😊

If there's a way I can help with the documentation effort (even just testing or reviewing), I'd be happy to contribute. Documentation is often overlooked but so important for OSS projects.

Thanks for all the work on Plane! 🙏

**Comment by sriramveeraghanta** [comment]
This has been already addressed in `v0.25.0` version. We don't have any plans to bring these variables back. So closing this issue.
